plugins/out_s3/s3.c

```diff
@@ -2202,7 +2274,7 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
                                       event_chunk->tag,
                                       flb_sds_len(event_chunk->tag));
             if (ret < 0) {
-                FLB_OUTPUT_RETURN(FLB_ERROR);
+                FLB_OUTPUT_RETURN(FLB_RETRY);
```

Comment:
I think the line below needs to be changed: https://github.com/Claych/fluent-bit/blob/clay-retry-limit-1.9/plugins/out_s3/s3.c#L2279

We should explicitly return `FLB_OUTPUT_RETURN(FLBOK);` for success. Otherwise we have a bug since I see in teh code that FLB_OK is not 0:

```
./include//fluent-bit/flb_macros.h:30:#define FLB_OK      1
```